lib/dependabot/metadata_finders/java/maven.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+require "dependabot/metadata_finders/base"
+
+module Dependabot
+  module MetadataFinders
+    module Java
+      class Maven < Dependabot::MetadataFinders::Base
+        private
+
+        def look_up_source
+          # TODO: return a `Source` object with a github, gitlab or bitbucket
```

Comment:
This is normally really easy. Are there API docs for the maven central registry? We're looking for something that allows us to get metadata about a dependency based on its name (specifically, we want to know where the source code is hosted).

Reply:
If no `<repositories> -> <repository>` is defined in the `pom.xml` (usually the root/aggregator pom), Maven Central will be used. If not, the specified repositories will be used. Potentially one must dynamically look up different repositories based on what the project is using.

I will have to look into how the Maven repositories work under the hood..

Reply:
Cool. For other languages we don't worry about providing metadata if a private repo is being used. If it's easy to do so then we totally should, but starting with Maven Central would be fine.